main/lsp/requests/hover.cc

```diff
@@ -48,20 +75,38 @@ LSPResult LSPLoop::handleTextDocumentHover(unique_ptr<core::GlobalState> gs, con
         }
 
         auto resp = move(queryResponses[0]);
+
+        optional<string> documentation = nullopt;
+        if (
+                resp->isIdent() || 
+                resp->isLiteral() || 
+                resp->isConstant() || 
+                resp->isDefinition()) {
+            auto loc = resp->getTypeAndOrigins().origins[0];
```

Comment:
Note: `getTypeAndOrigins().origins` could potentially be zero-length, so you should guard against this possibility.

Reply:
!! good catch 😅 a3e1eb6c4